### ISSUE TITLE
TypeScript: Last release broke delete method

### ISSUE THREAD

**Issue by 2-5** [vuex-orm/vuex-orm#442]
The last release broke code using `delete`:

```typescript
Setting.delete(record => record.name === 'x')
```

`Setting` is derived from `Model` and contains a `name` field. The error is:

    TS2339: Property 'name' does not exist on type 'Model'.

The problem seems to be that `Predicate` uses `Model`, instead of being templated to use `M`:

```typescript
export declare type Predicate = (model: Model) => boolean;

export default class Model {
  static delete<M extends typeof Model>(this: M, condition: Predicate): Promise<Collection<InstanceOf<M>>>;
}
```

**Comment by kiaking** [comment]
@2-5 Thank you so much for the report! I'll fix it!

**Comment by kiaking** [comment]
Fixed at #447. I'll be making a new release soon 🚀

**Comment by kiaking** [comment]
Sorry it took long. This issue is now fixed at Vuex ORM 0.32.3! 🔥 
https://github.com/vuex-orm/vuex-orm/releases/tag/v0.32.3
